iosApp/iosApp/feature/trial/TrialListView.swift

```diff
@@ -0,0 +1,107 @@
+            }
+            ScrollView(showsIndicators: false) {
+                LazyVGrid(columns: [GridItem(.adaptive(minimum: 180))], spacing: 8) {
+                    let trials = state?.trials.compactMap { $0 as? UITrialList.ItemTrial } ?? []
+                    Section(header:
+                        Text("Active Trials")
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                            .font(.system(size: 16, weight: .heavy))
+                    ) {
+                        ForEach(trials.filter({ !$0.data.trial.isRetired }), id: \.self) { trial in
+                            TrialJacket(trialData: trial.data)
+                        }
+                    }
+                    
+                    Section(header:
+                        Text("Retired Trials")
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                            .font(.system(size: 16, weight: .heavy))
+                    ) {
+                        ForEach(trials.filter({ $0.data.trial.isRetired }), id: \.self) { trial in
+                            TrialJacket(trialData: trial.data)
+                        }
+                    }
+                }
+            }
+        }
+        .onAppear {
+            viewModel.state.subscribe { state in
+                if let currentState = state {
+                    withAnimation {
+                        self.state = currentState
+                    }
+                }
+            }
+        }
+    }
+}
+
+struct PlacementBanner: View {
+    var banner: UIPlacementBanner
+    
+    var body: some View {
+        NavigationLink {
+            PlacementListView()
+        } label: {
+            HStack(spacing: 0) {
+                Text(banner.text.localized())
+                    .font(.system(size: 22, weight: .heavy))
+                    .padding(.trailing, 12)
+                    .foregroundColor(.white)
+                ForEach(banner.ranks, id: \.self) { rank in
+                    Image(String(describing: rank).lowercased())
+                        .resizable()
+                        .frame(width: 24.0, height: 24.0)
+                }
+            }
+        }
+        .padding()
+        .background(.gray).cornerRadius(10.0)
+    }
+}
+
+struct TrialJacket: View {
+    var trialData: UITrialJacket
+    
+    var body: some View {
+        Image(trialData.trial.name)
+            .resizable()
+            .aspectRatio(contentMode: .fill)
+            .overlay(alignment: .topLeading) { TrialDifficulty(difficulty: trialData.trial.difficulty as! Int) }
+    }
+}
+
+struct TrialDifficulty: View {
+    var difficulty: Int
+    
+    var body: some View {
+        Text(String(difficulty))
+            .font(.system(size: 24, weight: .bold))
+            .foregroundColor(.white)
+            .background(Circle().fill(Color.blue).frame(width: 40, height: 40).opacity(/*@START_MENU_TOKEN@*/0.8/*@END_MENU_TOKEN@*/))
```

Comment:
This should be broken into multiple lines for readability.

Reply:
Made a quick commit for this!